File: app/src/main/java/com/example/metasearch/manager/ImageAnalysisWorker.java
package com.example.metasearch.manager;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import com.example.metasearch.data.dao.DatabaseHelper;

public class ImageAnalysisWorker extends Worker {
    private ImageServiceRequestManager imageServiceRequestManager;
    private static final String TAG = "ImageAnalysisWorker";

    public ImageAnalysisWorker(
            @NonNull Context context,
            @NonNull WorkerParameters params) {
        super(context, params);
        DatabaseHelper databaseHelper = DatabaseHelper.getInstance(context);
        this.imageServiceRequestManager = ImageServiceRequestManager.getInstance(context,databaseHelper);
    }

    @NonNull
    @Override
    public Result doWork() {
        // 이미지 분석 시작
        Log.d(TAG, "ImageAnalysisWorker doWork() called");
        try {
            imageServiceRequestManager.getImagePathsAndUpload();
            Log.d(TAG, "ImageAnalysisWorker doWork() completed");
            return Result.success();
        } catch (Exception e) {
            Log.e(TAG, "ImageAnalysisWorker doWork() failed", e);
            return Result.failure();
        }
    }
}

File: feature/home/src/main/java/com/example/metasearch/feature/home/worker/ImageAnalysisWorker.kt
package com.example.metasearch.feature.home.worker

import android.content.Context
import androidx.work.CoroutineWorker
import androidx.work.WorkerParameters
import com.example.metasearch.core.data.api.repository.ImageAnalysisRepository
import com.example.metasearch.core.notification.notifier.AnalysisNotifier
import dagger.hilt.EntryPoint
import dagger.hilt.EntryPoints
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent

class ImageAnalysisWorker(
    context: Context,
    params: WorkerParameters
) : CoroutineWorker(context, params) {
    private val entryPoint = EntryPoints.get(applicationContext, AnalysisEntryPoint::class.java)
    private val repository = entryPoint.repository()
    private val notifier = entryPoint.notifier()

    override suspend fun doWork(): Result {
        return try {
            repository.runFullAnalysis()

            notifier.notifyComplete()

            Result.success()
        } catch (e: Exception) {
            Result.failure()
        }
    }

    @EntryPoint
    @InstallIn(SingletonComponent::class)
    interface AnalysisEntryPoint {
        fun repository(): ImageAnalysisRepository
        fun notifier(): AnalysisNotifier
    }
}

//@HiltWorker
//class ImageAnalysisWorker @AssistedInject constructor(
//    @Assisted context: Context,
//    @Assisted params: WorkerParameters,
//    private val imageAnalysisRepository: ImageAnalysisRepository,
//) : CoroutineWorker(context, params) {
//
//    override suspend fun doWork(): Result {
//        return try {
//            imageAnalysisRepository.runFullAnalysis()
//
//            val notificationWork = OneTimeWorkRequestBuilder<NotificationWorker>().build()
//
//            WorkManager.getInstance(applicationContext).enqueue(notificationWork)
//            Result.success()
//        } catch (e: Exception) {
//            Result.failure()
//        }
//    }
//}
